Corona-Warn-App/src/main/java/de/rki/coronawarnapp/risk/RiskLevelTask.kt

```diff
@@ -0,0 +1,135 @@
+package de.rki.coronawarnapp.risk
+
+import android.content.Context
+import de.rki.coronawarnapp.CoronaWarnApplication
+import de.rki.coronawarnapp.risk.RiskLevel.INCREASED_RISK
+import de.rki.coronawarnapp.risk.RiskLevel.LOW_LEVEL_RISK
+import de.rki.coronawarnapp.risk.RiskLevel.NO_CALCULATION_POSSIBLE_TRACING_OFF
+import de.rki.coronawarnapp.risk.RiskLevel.UNDETERMINED
+import de.rki.coronawarnapp.risk.RiskLevel.UNKNOWN_RISK_INITIAL
+import de.rki.coronawarnapp.risk.RiskLevel.UNKNOWN_RISK_OUTDATED_RESULTS
+import de.rki.coronawarnapp.risk.RiskLevel.UNKNOWN_RISK_OUTDATED_RESULTS_MANUAL
+import de.rki.coronawarnapp.storage.RiskLevelRepository
+import de.rki.coronawarnapp.task.Task
+import de.rki.coronawarnapp.task.TaskCancellationException
+import de.rki.coronawarnapp.task.TaskFactory
+import de.rki.coronawarnapp.task.common.DefaultProgress
+import de.rki.coronawarnapp.util.ConnectivityHelper
+import de.rki.coronawarnapp.util.ConnectivityHelper.isNetworkEnabled
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import org.joda.time.Duration
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Provider
+
+class RiskLevelTask @Inject constructor(
+    private val riskLevels: RiskLevels,
+    private val context: Context
```

Comment:
I think you need to annotate this with `@AppContext'

Reply:
The `context` needs to be annotated, not the class.
Because there are multiple contexts through out an Android app we want to tell dagger which specific one we want.
It's just a custom "tag" we created with the name "AppContext", which we apply in `AndroidModule.kt`.